locust/input_events.py

```diff
@@ -83,21 +93,32 @@ def get_poller():
         return UnixKeyPoller()
 
 
-def input_listener(key_to_func_map):
+def input_listener(key_to_user_params: Dict[str, Tuple[int, float]], runner: Runner):
     def input_listener_func():
         try:
             poller = get_poller()
         except InitError as e:
             logging.info(e)
             return
 
+        user_count = 0
         try:
             while True:
                 input = poller.poll()
                 if input:
-                    for key in key_to_func_map:
-                        if input == key:
-                            key_to_func_map[key]()
+                    try:
+                        user_params = key_to_user_params[input]
```

Comment:
We're losing a little flexibility for future development here. What if we want to add a new key press that does something completely different (like printing a status report or whatever - probably a useful feature btw :)

Reply:
I've made the `input_listener` more flexible by adding a second dictionary `key_to_callbacks`